ipv-core/src/main/java/com/ibm/whc/deid/util/ICDv10Manager.java

```diff
@@ -1,119 +1,125 @@
 /*
- * (C) Copyright IBM Corp. 2016,2020
+ * (C) Copyright IBM Corp. 2016,2021
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 package com.ibm.whc.deid.util;
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.Serializable;
+import java.security.SecureRandom;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
-import java.util.Map;
-
 import org.apache.commons.csv.CSVParser;
 import org.apache.commons.csv.CSVRecord;
-
-import com.ibm.whc.deid.models.ICD;
+import com.ibm.whc.deid.models.ICDWithoutFormat;
 import com.ibm.whc.deid.models.ICDFormat;
+import com.ibm.whc.deid.models.ICD;
 import com.ibm.whc.deid.shared.localization.Resource;
 import com.ibm.whc.deid.shared.localization.Resources;
 import com.ibm.whc.deid.util.localization.LocalizationManager;
 import com.ibm.whc.deid.util.localization.ResourceEntry;
 import com.ibm.whc.deid.utils.log.LogCodes;
 import com.ibm.whc.deid.utils.log.LogManager;
 
-/** The type Ic dv 10 manager. */
-public class ICDv10Manager implements Manager, Serializable {
-  /** */
-  private static final long serialVersionUID = -7207643604541499595L;
+/** The type ICDv10 Manager. */
+public class ICDv10Manager implements Manager {
 
-	protected final Collection<ResourceEntry> resourceICDList;
+  private static final LogManager logger = LogManager.getInstance();
 
-  protected final MapWithRandomPick<String, ICD> icdByCodeMap;
-  protected final MapWithRandomPick<String, ICD> icdByNameMap;
+  protected static final Resources resourceType = Resource.ICDV10;
 
-  protected int resourceInDbCount = 0;
+  private final SecureRandom random = new SecureRandom();
 
-  private static LogManager logger = LogManager.getInstance();
-  protected final Resources resourceType = Resource.ICDV10;
+  private final ArrayList<ICDWithoutFormat> icdList = new ArrayList<>(75000);
+  private final HashMap<String, ICDWithoutFormat> icdByCodeMap = new HashMap<>(100000);

```

Comment:
I guess you used the 41125 lines from the icd_v10 file and figured a little less than half for the list and a little more than 2x for the map ? It might be worth adding a comment there as to why those values and when we can expect to change given they are "hardcoded"

Reply:
Correct - There is a similar construction in the more "standard" managers where the actual constructor takes an "expectedCount" parameter.  For the ICD10 and ICD9, I'll look at creating a constant for "expectedCount" and then calculate appropriate sizes based on that, similar to what that constructor does.  I think I'll reduce the expected size to about 42000, though.  No need to take additional size until we actually update the CSVs.